R/convGDX2MIF.R

```diff
@@ -115,7 +115,17 @@ convGDX2MIF <- function(gdx, gdx_ref = NULL, file = NULL, scenario = "default",
   output <- add_dimension(output,dim=3.1,add = "model",nm = "REMIND")
   output <- add_dimension(output,dim=3.1,add = "scenario",nm = scenario)
 
+  message("Checking integrity of created MIF...")
 
+  sumChecks <- piamInterfaces::checkSummations(
+    mifFile = output, outputDirectory = NULL,
+    summationsFile = "extractVariableGroups",
+    absDiff = 1.5e-8, relDiff = 1e-8, roundDiff = FALSE
+  ) %>% filter(abs(!!sym("diff")) >= 1.5e-8)
+
+  if (nrow(sumChecks) > 0) {
+    warning("Summation checks have revealed some gaps! Run piamInterfaces::checkSummations")
+  }
```

Comment:
I wonder if it would be worthwhile to write the `checkSummations()` output to a file directly.
People will see that somewhere in the log file, but for most REMIND developers it won't be obvious what they should use `piamInterfaces::checkSummations()` on.
Just dumping this into some file would not cost us anything (kilobytes on gigabytes per run), but would greatly facilitate both debugging and ascertaining the validity of model output for projects.
```
  if (nrow(sumChecks) > 0) {
    if (is.null(file)) {
      warning("Summation checks have revealed some gaps! Run piamInterfaces::checkSummations")
    }
    else {
      summation_errors_file <- sub('(\\.[^.]+)$', '_summation_errors\\1', file)
      warning("Summation checks have revealed some gaps! See file ",
              summation_errors_file)
      write.table(sumChecks, summation_errors_file, quote = FALSE, sep = ';')
    }
  }
```

Reply:
Come to think of it, it would also be easier to just return the summation errors as an attribute (much like [`readr::problems()`](https://readr.tidyverse.org/reference/problems.html)), instead of telling people to run some extra code.
:point_right:  https://github.com/fbenke-pik/remind2/pull/1

Reply:
I agree. Except you could use `dataDumpFile` functionality of `checkSummations()` for that.

Reply:
Right, so unless @orichters wants to refactor checkSummations to only write to a file if some intolerable gaps were discovered, we go with Michaja's approach for now.